htdocs/manage/profile/index.bml

```diff
@@ -403,45 +405,64 @@ body<=
 
             $ret .= "<tr><td colspan=3><table summary=''>";
             my $oddeven = 0;
+            my $ct = 1;
 
-            foreach my $p (
-                           ["ao3", $ML{ '.services.ao3' }, 40],
-                           ["deviantart", $ML{'.services.deviantart'}, 20],
-                           ["diigo", $ML{ '.services.diigo' }, 16],
-                           ["discord", $ML{ '.services.discord' }, 40],
-                           ["etsy", $ML{ '.services.etsy' }, 20],
-                           ["ffnet", $ML{ '.services.ffnet' }, 30],
-                           ["github", $ML{'.services.github'}, 39],
-                           ["google_talk", $ML{'.chat.googlehangouts'}, 60],
-                           ["icq", $ML{'.chat.icquin'}, 12],
-                           ["insanejournal", $ML{'.services.insanejournal'}, 15],
-                           ["instagram", $ML{'.services.instagram'}, 30],
-                           ["jabber", $ML{'.chat.jabber'}, 60],
-                           ["last_fm_user", $ML{'.services.last_fm'}, 255],
-                           ["livejournal", $ML{'.services.livejournal'}, 30],
-                           ["medium", $ML{'.services.medium'}, 25],
-                           ["patreon", $ML{'.services.patreon'}, 255],
-                           ["pillowfort", $ML{'.services.pillowfort'}, 255],
-                           ["pinboard", $ML{'.services.pinboard'}, 30],
-                           ["pinterest", $ML{'.services.pinterest'}, 30],
-                           ["plurk", $ML{ '.services.plurk' }, 255],
-                           ["ravelry", $ML{ '.services.ravelry' }, 40],
-                           ["reddit", $ML{ '.services.reddit' }, 20],
-                           ["skype", $ML{'.chat.skype'}, 40],
-                           ["tumblr", $ML{ '.services.tumblr' }, 255],
-                           ["twitter", $ML{ '.services.twitter' }, 40],
-                           ["wattpad", $ML{ '.services.wattpad' }, 20],
-                          )
+            my $service_info = sub {
+                my ($site) = @_;
+                $site->{title} = LJ::Lang::ml( $site->{title_ml} );
+                return $site;
+            };
 
+            my @services = map $service_info->($_), @{ DW::External::ProfileServices->list };
+            my @dropdown = ( '' => '' );
+            push @dropdown, ( $_->{name} => $_->{title} ) foreach @services;
+
+            my $service_td = sub {
+                my ( $p, $val ) = @_;
+                $oddeven = !$oddeven;
+                $ret .= "<tr class='field_block'>" if $oddeven;
+                $ret .= "<td class='field_name' width='20%'>$p->{title}</td><td width='30%'>";
+                $ret .= LJ::html_hidden( "extservice_name_$ct", $p->{name} );
```

Comment:
this puts an implicit requirement on the name field that it can only be "stuff that's safe to put in a field name." So, no hyphens or periods, other symbols. we should call this out up in the database schema/inserts that the name field has to be `[a-z0-9_]` or something.